sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotCheckZeroSizeCollection.cs

```diff
@@ -0,0 +1,151 @@
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)c.Node;
+                    CheckCondition(c, binaryExpression.Left, binaryExpression.Right);
+                }, SyntaxKind.GreaterThanOrEqualExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)c.Node;
+                    CheckCondition(c, binaryExpression.Right, binaryExpression.Left);
+                }, SyntaxKind.LessThanOrEqualExpression);
+        }
+
+        private static void CheckCondition(SyntaxNodeAnalysisContext context, ExpressionSyntax expressionValueNode,
+            ExpressionSyntax constantValueNode)
+        {
+            if (!IsConstantZero(context, constantValueNode))
+            {
+                return;
+            }
+
+            ISymbol symbol = GetSymbol(context, expressionValueNode);
+            if (symbol == null)
+            {
+                return;
+            }
+
+            string symbolType = GetDeclaringTypeName(symbol);
+            if (symbolType == null)
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(Diagnostic.Create(rule, context.Node.GetLocation(),
+                symbol.Name.ToLower(), symbolType));
+        }
+
+        private static ISymbol GetSymbol(SyntaxNodeAnalysisContext context, ExpressionSyntax expression)
+        {
+            while (true)
+            {
+                var conditionalAccess = expression as ConditionalAccessExpressionSyntax;
+                if (conditionalAccess == null)
+                {
+                    break;
+                }
+                expression = conditionalAccess.WhenNotNull;
+            }
+
+            return context.SemanticModel.GetSymbolInfo(expression).Symbol;
+        }
+
+        private static string GetDeclaringTypeName(ISymbol symbol)
+        {
+            if (IsArrayLengthProperty(symbol))
+            {
+                return "Array";
+            }
+
+            if (IsEnumerableCountMethod(symbol))
+            {
+                return "IEnumerable<T>";
+            }
+
+            if (IsCollectionProperty(symbol))
+            {
+                return "ICollection";
+            }
+
+            return null;
+        }
+
+        private static bool IsConstantZero(SyntaxNodeAnalysisContext context, ExpressionSyntax expression)
+        {
+            var constantExpressionNode = expression.RemoveParentheses();
+            Optional<object> constant = context.SemanticModel.GetConstantValue(constantExpressionNode);
+            if (!constant.HasValue)
+            {
+                return false;
+            }
+
+            return (constant.Value is int) ? (int)constant.Value == 0 : false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1125)